src/groupeddatatable/grouping.jl

```diff
@@ -236,12 +236,28 @@ colwise(sum, groupby(dt, :a))
 ```
 
 """
-colwise(f::Function, d::AbstractDataTable) = map(i -> f(d[i]), 1:ncol(d))
+function colwise(f::Function, d::AbstractDataTable)
+    x = [f(d[i]) for i in 1:ncol(d)]
+    if eltype(x) <: Nullable && isa(x, Vector{eltype(x)})
+        return NullableArray(x)
+    end
+    return x
+end
 colwise(f::Function, gd::GroupedDataTable) = map(colwise(f), gd)
 colwise(f::Function) = x -> colwise(f, x)
 colwise(f) = x -> colwise(f, x)
 # apply several functions to each column in a DataTable
-colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable) = map(i -> map(f -> f(d[i]), fns), 1:ncol(d))
+function colwise{T<:Function}(fns::Vector{T}, d::AbstractDataTable)
+    res = Array{AbstractVector}(ncol(d))
```

Comment:
I'm trying to think of a situation in which it may actually make sense to have functions that result in different sizes and I'm coming up short. In general I think a `Matrix` is much more natural to work with than an array of (abstract)arrays. We'd just have to make sure that it's a `Matrix{Any}` to avoid any potentially unwanted promotion due to the functions returning different types.